app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/BusinessVerificationService.scala

```diff
@@ -17,18 +17,35 @@
 package uk.gov.hmrc.incorporatedentityidentificationfrontend.services
 
 import javax.inject.{Inject, Singleton}
-import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException}
 import uk.gov.hmrc.incorporatedentityidentificationfrontend.connectors.{CreateBusinessVerificationJourneyConnector, RetrieveBusinessVerificationStatusConnector}
-import uk.gov.hmrc.incorporatedentityidentificationfrontend.models.BusinessVerificationStatus
+import uk.gov.hmrc.incorporatedentityidentificationfrontend.models.{BusinessVerificationFail, BusinessVerificationStatus, BusinessVerificationUnchallenged, NotEnoughEvidence, UserLockedOut}
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class BusinessVerificationService @Inject()(createBusinessVerificationJourneyConnector: CreateBusinessVerificationJourneyConnector,
-                                            retrieveBusinessVerificationResultConnector: RetrieveBusinessVerificationStatusConnector) {
+                                            retrieveBusinessVerificationResultConnector: RetrieveBusinessVerificationStatusConnector,
+                                            incorporatedEntityInformationService: IncorporatedEntityInformationService)(implicit val executionContext: ExecutionContext) {
 
   def createBusinessVerificationJourney(journeyId: String, ctutr: String)(implicit hc: HeaderCarrier): Future[Option[String]] =
-    createBusinessVerificationJourneyConnector.createBusinessVerificationJourney(journeyId, ctutr)
+    createBusinessVerificationJourneyConnector.createBusinessVerificationJourney(journeyId, ctutr).flatMap {
+      eitherRedirect =>
+        eitherRedirect.fold(
```

Comment:
Instead of a fold, you can just do a pattern match directly on the either.

If you flatMap, then within the braces you can go:
`case Right(JourneyCreated(redirectUrl) => ???`
`case Left(NotEnoughEvidence) => ???`
`case Left(UserLockedOut) => ???`

Or if you want to handle both lefts within one case, you can just assign that to a variable and match on it afterwards:
`case Left(failureReason) =>`
`val bvStatus = journeyCreationFailed match {`
              `case NotEnoughEvidence => BusinessVerificationUnchallenged`
              `case UserLockedOut => BusinessVerificationFail`
             ` case _ => throw new InternalServerException(s"createBusinessVerificationJourney service failed with invalid BV status")`

Your code looks like it will work fine, I just really have a thing against folds...

Reply:
I've done this